apps/server/src/shared/infra/mail/mail.service.ts

```diff
@@ -16,6 +17,39 @@ export class MailService {
 	) {}
 
 	public async send(data: Mail): Promise<void> {
+		if (Configuration.has('ADDITIONAL_BLACKLISTED_EMAIL_DOMAINS')) {
+			const domainBlockList = Configuration.get('ADDITIONAL_BLACKLISTED_EMAIL_DOMAINS') as string;
+			data.recipients = this.removeEmailAddressesThatHaveBlockedDomain(data.recipients, domainBlockList) as string[];
+			data.cc = this.removeEmailAddressesThatHaveBlockedDomain(data.cc, domainBlockList);
+			data.bcc = this.removeEmailAddressesThatHaveBlockedDomain(data.bcc, domainBlockList);
+			data.replyTo = this.removeEmailAddressesThatHaveBlockedDomain(data.replyTo, domainBlockList);
+		}
+
+		if (data.recipients.length === 0) {
+			return;
+		}
+
 		await this.amqpConnection.publish(this.options.exchange, this.options.routingKey, data, { persistent: true });
 	}
+
+	private removeEmailAddressesThatHaveBlockedDomain(
+		mails: string[] | undefined,
+		domainBlockList: string
+	): string[] | undefined {
+		if (mails === undefined || mails === null) {
+			return mails;
+		}
+		const domainBlockListArray = domainBlockList.split(',').map((domain) => domain.trim());
+
+		for (const blockedDomain of domainBlockListArray) {
+			// eslint-disable-next-line no-plusplus
+			for (let i = mails.length - 1; i > -1; i--) {
+				const mailDomain = mails[i].split('@')[1];
+				if (mailDomain === blockedDomain) {
+					mails.splice(i, 1);
```

Comment:
Please do not modify the array you're iterating through - it works here, but I think it's generally an anti-pattern - please create a new empty array first and simply add all the email addresses that are not containing a blacklisted domain.

Reply:
done